packages/astro/src/types/public/config.ts

```diff
@@ -1864,6 +1927,36 @@ export interface ViteUserConfig extends OriginalViteUserConfig {
 		 */
 
 		responsiveImages?: boolean;
+
+		/**
+		 * @docs
+		 * @name experimental.sessions
+		 * @type {boolean}
+		 * @default `false`
+		 * @version 5.0.0
+		 * @see session
+		 * @description
+		 * Enables support for sessions in Astro. When enabled you can access the `Astro.session` object to read and write data that persists across requests. You can configure the session driver using the `session` option, or use the default provided by the adapter.
+		 *
+		 * ```js title=astro.config.mjs
+		 * {
+		 * 	experimental: {
+		 * 		sessions: true,
+		 * 	},
+		 * }
+		 * ```
+		 *
+		 * ```astro title=src/components/CartButton.astro
+		 * ---
+		 * const cart = await Astro.session.get('cart');
```

Comment:
```suggestion
		 * Enables support for sessions in Astro. Sessions are used to store user data across requests, such as user authentication state.
		 * 
		 * When enabled you can access the `Astro.session` object to read and write data that persists across requests. You can configure the session driver using the [`session` option](#session), or use the default provided by your adapter.
		 *
		 * ```js title=astro.config.mjs
		 * {
		 * 	experimental: {
		 * 		sessions: true,
		 * 	},
		 * }
		 * ```
		 *
		 * ```astro title=src/components/CartButton.astro
		 * ---
		 * export const prerender = false; // Not needed in 'server' mode
		 * const cart = await Astro.session.get('cart');
```
Minor suggestions here:
- THIS is the first place people will go for the feature, so this is probably where we should say what sessions are (as opposed to up in session config, or at least in addition to)
- separate paragraphs for easier readability
- linking to the config section
- just checking: if this requires an adapter, then does it also require on-demand rendering on the page being used? If so, we'd normally show the extra line in the code sample.